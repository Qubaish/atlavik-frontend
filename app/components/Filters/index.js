/**
 *
 * Img.js
 *
 * Renders an image, enforcing the usage of the alt="" tag
 */

import React, {Fragment} from 'react';
import PropTypes from 'prop-types';
import Input from 'components/Input';
import DropDown from 'components/DropDown';
import Range from 'components/Range';
import DateRange from 'components/DateRange';

function Filters({type, state, onChange}) {
    switch (type) {
        case 'name':
            return (
                <Input val={state.name} onChange={onChange}/>
            );
        case 'city':
            return (
                <DropDown val={state.city} onChange={onChange}/>
            );
        case 'price':
            return (
                <Range val={state.price} onChange={onChange}/>
            );
        case 'date':
            return (
                <DateRange val={state.date} onChange={onChange}/>
            );
        case 'all':
            return (
                <Fragment>
                    <Input val={state.name} onChange={onChange}/>
                    <DropDown val={state.city} onChange={onChange}/>
                    <Range val={state.price} onChange={onChange}/>
                    <DateRange val={state.date} onChange={onChange}/>
                </Fragment>
            );
        default:
            return '';
    }
}

// We require the use of src and alt, only enforced by react in dev mode
Filters.propTypes = {
    type: PropTypes.string.isRequired,
};

export default Filters;
