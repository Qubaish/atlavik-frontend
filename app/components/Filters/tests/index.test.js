import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Filters from '../index';

describe('<Filters />', () => {
    it('should render a Input', () => {
        const container = shallow(
            <Filters type="name" state={{name: 'val'}} onChange={jest.fn()} />
        );
        expect(shallowToJson(container)).toMatchSnapshot();
    });
    it('should render a Input', () => {
        const container = shallow(
            <Filters type="city" state={{city: 'val'}} onChange={jest.fn()} />
        );
        expect(shallowToJson(container)).toMatchSnapshot();
    });
    it('should render a Input', () => {
        const container = shallow(
            <Filters type="price" state={{price: {min: 0, max: 20}}} onChange={jest.fn()} />
        );
        expect(shallowToJson(container)).toMatchSnapshot();
    });
    it('should render a Input', () => {
        const container = shallow(
            <Filters type="date" state={{date: ['15-10-2020', '15-10-2020']}} onChange={jest.fn()} />
        );
        expect(shallowToJson(container)).toMatchSnapshot();
    });
    it('should render a Input', () => {
        const container = shallow(
            <Filters type="all" state={{name: 'val', city: 'dubai'}} onChange={jest.fn()} />
        );
        expect(shallowToJson(container)).toMatchSnapshot();
    });
});
