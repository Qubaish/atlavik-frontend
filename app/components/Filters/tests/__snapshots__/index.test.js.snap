// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> should render a Input 1`] = `
<Input
  onChange={[MockFunction]}
  val="val"
/>
`;

exports[`<Filters /> should render a Input 2`] = `
<DropDown
  onChange={[MockFunction]}
  val="val"
/>
`;

exports[`<Filters /> should render a Input 3`] = `
<Range
  onChange={[MockFunction]}
  val={
    Object {
      "max": 20,
      "min": 0,
    }
  }
/>
`;

exports[`<Filters /> should render a Input 4`] = `
<DateRange
  onChange={[MockFunction]}
  val={
    Array [
      "15-10-2020",
      "15-10-2020",
    ]
  }
/>
`;

exports[`<Filters /> should render a Input 5`] = `
<Fragment>
  <Input
    onChange={[MockFunction]}
    val="val"
  />
  <DropDown
    onChange={[MockFunction]}
    val="dubai"
  />
  <Range
    onChange={[MockFunction]}
  />
  <DateRange
    onChange={[MockFunction]}
  />
</Fragment>
`;
